lib/events/auditlog.go

```diff
@@ -890,7 +901,37 @@ func (l *AuditLog) SearchSessionEvents(fromUTC, toUTC time.Time) ([]EventFields,
 		SessionEndEvent,
 	}
 
-	return l.SearchEvents(fromUTC, toUTC, query.Encode())
+	// because of the limit and distributed nature of auth server event
+	// logs, some events can be fetched with session end event and without
+	// session start event. to fix this, the code below filters out the events without
+	// start event to guarantee that all events in the range will get fetched
+	events, err := l.SearchEvents(fromUTC, toUTC, query.Encode(), limit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// filter out events that have start event, but don't have end event
+	started := make(map[string]bool, len(events)/2)
```

Comment:
Using `map[string]struct{}` for things where value does not matter is a bit more efficient.